dobson/utils.py

```diff
@@ -62,3 +68,33 @@ def get_mac_addresses():
 
         mac_addr = device[-18:].strip(' ').lower().replace(' ', ':')
         yield mac_addr
+
+
+def get_unknown_mac_addresses():
+    """
+    Yields MAC Addresses connected to the router which are _not_ in devices.json. Useful for
+    tagging new devices
+    """
+    for mac_addr in get_mac_addresses():
+        if mac_addr not in KNOWN_DEVICES:
+            yield mac_addr
+
+
+def add_device(mac_addr, data):
+    """Adds an entry for the specified Mac Address to devices.json and KNOWN_DEVICES
+    If the specified Mac Address already exists, returns False and does not update"""
+    mac_addr = mac_addr.lower()
+    if mac_addr in KNOWN_DEVICES:
+        return False
```

Comment:
I'd probably load in the file and make sure then that the mac address isn't present first, since the file could have changed but the process hasn't been restarted, something like that. Maybe checking the file's last modified time would be reasonable to see whether it's been updated more recently so that even when a query is done a reload can be performed? Idk, just throwing out possible ideas